openmc/deplete/chain.py

```diff
@@ -703,6 +704,65 @@ def form_matrix(self, rates, fission_yields=None):
         # Return CSC representation instead of DOK
         return matrix.tocsc()
 
+    def add_redox_term(self, matrix, buffer, oxidation_states):
+        """Adds a redox term to the depletion matrix from data contained in
+        the matrix itself and a few user-inputs.
+
+        The redox term to add to the buffer nuclide :math:`N_b` can be written
+        as: :math:`\frac{dN_b(t)}{dt} =
+                \cdots + \frac{1}{Ox_b}\sum_i N_i\left( L_{ii}Ox_i -
+                \sum_j G_{i\rightarrow j } Ox_j\right)`
+
+        where :math:`Ox_b` and :math:`Ox_j` are the oxidation states for the
+        corresponding buffer elmenent and j-th nuclide.
+        The first term in the right hand side represent the losses in the
+        diagonal terms of the Bateman matrix, for each nuclide :math:`i`, and
+        the second one the gains in the off-diagonal terms, multiplied by their
+        respective oxidation states.
+
+        Parameters
+        ----------
+        matrix : scipy.sparse.csr_matrix
```

Comment:
OpenMC now uses CSC, which was found to be more numerically stable when used in conjunction with scipy.sparse functionality.